src/crimson/osd/osd_operations/snaptrim_event.h

```diff
@@ -37,10 +37,9 @@ class SnapTrimEvent final : public PhasedOperationT<SnapTrimEvent> {
       IOInterruptCondition, remove_or_update_ertr>;
   using snap_trim_ertr = remove_or_update_ertr::extend<
     crimson::ct_error::eagain>;
-  using snap_trim_iertr = remove_or_update_iertr::extend<
-    crimson::ct_error::eagain>;
+  using snap_trim_iertr = remove_or_update_iertr;
   using snap_trim_event_ret_t =
-    snap_trim_ertr::future<seastar::stop_iteration>;
```

Comment:
Is `snap_trim_ertr` needed? It seems like the last user is removed, if so we can cleanup.